### PR TITLE
build(py2): Drop Python 2 support in CI

### PR DESCRIPTION
**Pull request by armenzg**
This makes all CI workflows run Python 3.6.10 and removes all executions of Python 2.
This does *not* drop Python 2 support.

We can follow up soon after. We can land this after the release tomorrow.

### CODE REVIEW

**Comment by armenzg** [comment]
This is ready to review, even though, I'm looking at fixing the tests for Plugins and Symbolicator since they were not running on Python 3.

Please review while I tackle those.

**Review by BYK — COMMENTED** [review]
I'm r+ but we need to remove all Python 2 support from getsentry/onpremise first. Otherwise people who run PY2 builds may see errors.

Also, I think we should have one more Py2 release with 21.1.0 and then clearly announce we are dropping Py2 support to give some warning. What do you think?

File: .craft.yml
Comment (chadwhitacre) [inline .craft.yml:27]: Does `py2` still belong in this string?

Comment (armenzg) [reply .craft.yml:27]: I wasn't sure. I got the feeling that it's simply the naming for wheels that are compatible with Python 2 & 3.
I've seen those filenames in the output of pip installations.

**Comment by armenzg** [comment]
> I'm r+ but we need to remove all Python 2 support from getsentry/onpremise first. Otherwise people who run PY2 builds may see errors.
> 
> Also, I think we should have one more Py2 release with 21.1.0 and then clearly announce we are dropping Py2 support to give some warning. What do you think?

Are you suggesting we wait until the 15th/16th? I'm fine as long as there are no objections from the folks on `#proj-delete-python2`

Can I assume you're handling the `onpremise` work?

**Comment by chadwhitacre** [comment]
I find a number of references to `[^a-z]py2[^a-z]` still. Are these intentionally left in?

```
tests/acceptance/test_emails.py:46:    version_suffix = "_py2" if six.PY2 else ""
tests/sentry/plugins/bases/notify/tests.py:32:        # XXX(py3): Handle ordering differences between py2/3
tests/sentry/web/frontend/test_oauth_authorize.py:149:        # between py2/3
tests/sentry/utils/hashlib/tests.py:23:    # and not the text_type, since they will map to different cases (py2 will
tests/sentry/utils/hashlib/tests.py:35:    if six.PY2:
src/sentry/tsdb/redis.py:200:                # XXX: Python 3 reprs bytes differently to Python 2. In py2,
src/sentry/runner/commands/devservices.py:55:    # Convert ns -> s, float in both py2 + 3.
src/sentry/utils/compat/__init__.py:37:if six.PY2:
src/sentry/data_export/tasks.py:115:                # module is able to do, it will NOT write bytes like in py2).
src/sentry/data_export/tasks.py:118:                if six.PY2:
src/sentry/data_export/tasks.py:240:    if six.PY2:
src/sentry/data_export/tasks.py:252:    if six.PY2:
src/sentry/monkey/pickle.py:60:# [0]: https://rebeccabilbro.github.io/convert-py2-pickles-to-py3/#python-2-objects-vs-python-3-objects
```

File: .craft.yml
Comment (BYK) [reply .craft.yml:27]: Yes as we generate universal bundles. We should change that too, good catch.

**Comment by BYK** [comment]
>Are you suggesting we wait until the 15th/16th? I'm fine as long as there are no objections from the folks on #proj-delete-python2

Yup.

>Can I assume you're handling the onpremise work?

Well, we expect teams to handle on-premise work related to their projects themselves but since I am part of the PY3 team, I'll get that part done 🙂 Just, not until 15th.

**Comment by armenzg** [comment]
> I find a number of references to `[^a-z]py2[^a-z]` still. Are these intentionally left in?
> 
> ```
> tests/acceptance/test_emails.py:46:    version_suffix = "_py2" if six.PY2 else ""
> tests/sentry/plugins/bases/notify/tests.py:32:        # XXX(py3): Handle ordering differences between py2/3
> tests/sentry/web/frontend/test_oauth_authorize.py:149:        # between py2/3
> tests/sentry/utils/hashlib/tests.py:23:    # and not the text_type, since they will map to different cases (py2 will
> tests/sentry/utils/hashlib/tests.py:35:    if six.PY2:
> src/sentry/tsdb/redis.py:200:                # XXX: Python 3 reprs bytes differently to Python 2. In py2,
> src/sentry/runner/commands/devservices.py:55:    # Convert ns -> s, float in both py2 + 3.
> src/sentry/utils/compat/__init__.py:37:if six.PY2:
> src/sentry/data_export/tasks.py:115:                # module is able to do, it will NOT write bytes like in py2).
> src/sentry/data_export/tasks.py:118:                if six.PY2:
> src/sentry/data_export/tasks.py:240:    if six.PY2:
> src/sentry/data_export/tasks.py:252:    if six.PY2:
> src/sentry/monkey/pickle.py:60:# [0]: https://rebeccabilbro.github.io/convert-py2-pickles-to-py3/#python-2-objects-vs-python-3-objects
> ```

I was hoping some devs would pick up this part since it would require looking at the product more closely. I'm also assuming that if I get the CI green w/o touching these it probably means we're not testing for it.

File: .github/workflows/acceptance.yml
Comment (billyvg) [inline .github/workflows/acceptance.yml:95]: Can we do what we are doing in `acceptance-py3.6.yml` where it reads from `.python-version` file?

**Comment by billyvg** [comment]
Would it be better to split this up into 1) CI, 2) Dev-env, 3) release?

**Comment by armenzg** [comment]
I'm moving the dev env part of it to #22903 since we can land it in advance and makes reviewers burden lighter.

**Comment by armenzg** [comment]
Moving to draft until PR #22903 is merged and we will resume this closer to the 15th.

**Comment by armenzg** [comment]
I've rebased this PR with the dev env changes that just merged.

> Would it be better to split this up into 1) CI, 2) Dev-env, 3) release?

This PR contains CI & release work. I would think we should keep this two together until the 15th. Thoughts?

File: .github/workflows/acceptance.yml
Comment (armenzg) [reply .github/workflows/acceptance.yml]: Addressed.

**Comment by armenzg** [comment]
For anyone following, I'm only focusing on CI changes and *not* on dropping the hard check of Python 2.7 not being accepted. We can follow up with that soon after.

File: .github/workflows/acceptance.yml
Comment (billyvg) [inline .github/workflows/acceptance.yml:75]: So we will need to update GH branch settings to reflect the new name. We should drop `python` as it's not relevant.

```suggestion
    name: acceptance
```

File: .github/workflows/acceptance.yml
Comment (billyvg) [inline .github/workflows/acceptance.yml:95]: Do we need to grep `3.6` here? Wouldn't it be better to use whatever is in `.python-version`, or at least the first line?

**Review by billyvg — COMMENTED** [review]
Let's remove all `py3.6` from names since we're not testing multiple versions and 3.6 will be considered the supported version.

File: .github/workflows/acceptance.yml
Comment (joshuarli) [inline .github/workflows/acceptance.yml:83]: This isn't needed for acceptance, only backend testing where we want to explicitly run django migrations.

File: .github/workflows/acceptance.yml
Comment (joshuarli) [reply .github/workflows/acceptance.yml:95]: @armenzg Let's just use the first line throughout everything now.

**Review by joshuarli — COMMENTED** [review]
Looks good to me, after a few comments are addressed. Will leave final approval to Billy.

File: tests/sentry_plugins/sessionstack/test_plugin.py
Comment (billyvg) [inline tests/sentry_plugins/sessionstack/test_plugin.py:12]: Can we revert this change?

**Review by billyvg — APPROVED** [review]
This looks great, before merging, we'll need to update the GH branch settings for the new `acceptance` job name, and we'll need to warn devs that they'll need to rebase to run the updated workflows.

File: tests/sentry_plugins/sessionstack/test_plugin.py
Comment (armenzg) [reply tests/sentry_plugins/sessionstack/test_plugin.py:12]: I tried reverting it, however, the tests fail.
I believe that moving to Python 3 installs a different dependency for whatever handles URL query parameters.
It seems to me only a change in order of parameters.

File: .github/workflows/acceptance.yml
Comment (billyvg) [reply .github/workflows/acceptance.yml]: It doesn't affect Freight because this is on `sentry` repo, it will affect GitHub settings (merging PRs)
image

**Comment by billyvg** [comment]
@joshuarli Do the visual snapshot diffs line-up to what you experienced previously?

**Comment by joshuarli** [comment]
> @joshuarli Do the visual snapshot diffs line-up to what you experienced previously?

backlink: https://github.com/getsentry/sentry/pull/21098#issuecomment-703812194

I took a look, they're all just RNG. And looks like the problematic one (`acceptance/organization-releases-with-releases.png`) is gone.

So I approved the visual snapshot.

**Comment by armenzg** [comment]
Waiting on @BYK 's green light and for the "required checks" to be removed (I can't do it myself).

**Comment by BYK** [comment]
Blocked by https://github.com/getsentry/onpremise/pull/833

**Comment by BYK** [comment]
LGTM. Would you like me to follow up with `setup.py` and `cloudbuild.yml` changes or do you wanna tuck them into this PR @armenzg?

**Review by BYK — APPROVED** [review]

**Comment by armenzg** [comment]
Woohoo!

**Pull request merged**
